tests/core/test_rule_groups.py

```diff
@@ -0,0 +1,38 @@
+from smdebug_rulesconfig.debugger_rules._constants import (
+    SUPPORTED_DL_FRAMEWORKS,
+    SUPPORTED_FRAMEWORKS,
+)
+from smdebug_rulesconfig.debugger_rules._ruleGroups import (
+    DEEP_LEARNING_APPLICATION_RULES,
+    DEEP_LEARNING_RULES,
+    UNIVERSAL_RULES,
+    XGBOOST_RULES,
+)
+from smdebug_rulesconfig.debugger_rules._utils import _get_rule_list
+
+
+def test_universal_rules():
+    for rule in UNIVERSAL_RULES:
+        for framework in SUPPORTED_FRAMEWORKS:
+            if rule not in _get_rule_list(framework):
+                assert False, f"Rule: {rule} not supported by {framework}"
+
+
+def test_deep_learning_rules():
+    for rule in DEEP_LEARNING_RULES:
+        for framework in SUPPORTED_DL_FRAMEWORKS:
+            if rule not in _get_rule_list(framework):
+                assert False, f"Rule: {rule} not supported by {framework}"
+
+
+def test_deep_learning_application_rules():
+    for rule in DEEP_LEARNING_APPLICATION_RULES:
```

Comment:
is there a reason why we need to make a distinction between DEEP_LEARNING_RULES and DEEP_LEARNING_APPLICATION_RULES?

Reply:
We've made the distinction on the docs page: https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-built-in-rules.html

I simply followed the same convention.